protobuf-converter/src/main/java/io/confluent/connect/protobuf/ProtobufConverter.java

```diff
@@ -0,0 +1,162 @@
+import io.confluent.kafka.serializers.protobuf.AbstractKafkaProtobufDeserializer;
+import io.confluent.kafka.serializers.protobuf.AbstractKafkaProtobufSerializer;
+import io.confluent.kafka.serializers.protobuf.KafkaProtobufDeserializerConfig;
+import io.confluent.kafka.serializers.protobuf.KafkaProtobufSerializerConfig;
+import io.confluent.kafka.serializers.protobuf.ProtobufSchemaAndValue;
+
+
+/**
+ * Implementation of Converter that uses Protobuf schemas and objects.
+ */
+public class ProtobufConverter implements Converter {
+
+  private SchemaRegistryClient schemaRegistry;
+  private Serializer serializer;
+  private Deserializer deserializer;
+
+  private boolean isKey;
+  private ProtobufData protobufData;
+
+  public ProtobufConverter() {
+  }
+
+  // Public only for testing
+  public ProtobufConverter(SchemaRegistryClient client) {
+    schemaRegistry = client;
+  }
+
+  @Override
+  public void configure(Map<String, ?> configs, boolean isKey) {
+    this.isKey = isKey;
+    ProtobufConverterConfig protobufConverterConfig = new ProtobufConverterConfig(configs);
+
+    if (schemaRegistry == null) {
+      schemaRegistry =
+          new CachedSchemaRegistryClient(protobufConverterConfig.getSchemaRegistryUrls(),
+          protobufConverterConfig.getMaxSchemasPerSubject(),
+          configs
+      );
+    }
+
+    serializer = new Serializer(configs, schemaRegistry);
+    deserializer = new Deserializer(configs, schemaRegistry);
+    protobufData = new ProtobufData(new ProtobufDataConfig(configs));
+  }
+
+  @Override
+  public byte[] fromConnectData(String topic, Schema schema, Object value) {
+    try {
+      ProtobufSchemaAndValue schemaAndValue = protobufData.fromConnectData(schema, value);
+      Object v = schemaAndValue.getValue();
+      if (v instanceof Message) {
+        return serializer.serialize(topic,
+            isKey,
+            (Message) v,
+            schemaAndValue.getSchema()
+        );
+      } else {
+        throw new DataException("Unsupported object of class " + v.getClass().getName());
+      }
+    } catch (SerializationException e) {
+      throw new DataException(String.format(
+          "Failed to serialize Protobuf data from topic %s :",
+          topic
+      ), e);
+    }
+  }
+
+  @Override
+  public SchemaAndValue toConnectData(String topic, byte[] value) {
+    try {
+      ProtobufSchemaAndValue deserialized = deserializer.deserialize(topic, isKey, value);
+
+      if (deserialized == null || deserialized.getValue() == null) {
+        return SchemaAndValue.NULL;
+      } else {
+        Object object = deserialized.getValue();
+        if (object instanceof DynamicMessage) {
+          DynamicMessage message = (DynamicMessage) object;
+          return protobufData.toConnectData(deserialized.getSchema(), message);
+        }
+      }
+      throw new DataException(String.format("Unsupported type returned during deserialization of "
+              + "topic %s ",
+          topic
```

Comment:
Question: how do we decide in which case we use String.format and in which case we just 'add' them together via the '+' operator ?

Reply:
I'll remove the `+`, that was added by the IDE because the line was too long

Reply:
My bad, what I meant was: how do I decide between these two ?

`throw new Exception("Unsupported type for topic " + topic);`

and

`throw new Exception(String.format("Unsupported type for topic %s", topic));`